src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java

```diff
@@ -340,4 +340,12 @@ public void draggedOn(GroupNodeViewModel target, DroppingMouseLocation mouseLoca
     private int getPositionInParent() {
         return groupNode.getPositionInParent();
     }
+
+    public void setDisplayItemsCount(boolean displayItemsCount) {
+        this.displayItemsCount = displayItemsCount;
+
+        if (displayItemsCount) {
+            calculateNumberOfMatches();
+        }
```

Comment:
Calculating count of items here, instead of constructor. 
Next time they are calculated when "onDatabaseChanged" method fires.